src/test/java/selenium/pageobjects/WebLabAnswerPage.java

```diff
@@ -0,0 +1,61 @@
+package selenium.pageobjects;
+
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+
+import java.awt.*;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.UnsupportedFlavorException;
+import java.io.IOException;
+
+public class WebLabAnswerPage extends BasePageObject {
+
+    public WebLabAnswerPage(WebDriver driver, String url) {
+        super(driver, url);
+    }
+
+    @FindBy(xpath = "/html/body/div[2]/div[5]/div/div/div[3]/div[1]/div/div/div[1]")
+    private WebElement solutionDiv;
+
+    @FindBy(xpath = "/html/body/div[2]/div[5]/div/div/div[3]/div[1]/div/div/a[1]")
+    private WebElement unlockAnswerButton;
+
+    public String getSolution() {
+        this.awaitPageLoaded();
+
+        // unlock answer if it is not yet unlocked
+        if (this.isDisplayed(unlockAnswerButton)) {
+            this.unlockAnswerButton.click();
+            this.awaitElementVisibility(this.solutionDiv);
+        }
+
+        this.solutionDiv.click();
```

Comment:
Ahh, I see why you clear up the clipboard. 

I _would not_ do this this way. Just get the content via selenium, and put it in a variable. Why are you copying and pasteing?

Reply:
Because the solution isn't contained in a single element and the editor dynamically changes the contents of the elements, so there's no easy way to retrieve the solution apart from copying and pasting. It is indeed a dirty solution and it turns out it doesn't work on a headless browser, so we can't use it. I'll look into other ways to retrieve the solution, maybe via JavaScript.

Reply:
@mauricioaniche Using JS worked flawlessly! :) In fact, using JS is much faster than typing the solution into the editor, so I implemented it in the submission code as well, which made the tests about 20% faster.